bucket_test.go

```diff
@@ -464,6 +464,217 @@ func TestBucket_Delete_NonExisting(t *testing.T) {
+		},
+		{
+			"subBucket key exist in srcBucket, but no subBucket value",
+			"srcBucket",
+			"subBucket",
+			"this is subBucket key",
+			"this is subBucket value",
+			"dstBucket",
+			false,
+			false,
+			true,
+			false,
+			berrors.ErrIncompatibleValue,
+		},
+		{
+			"subBucket key exist in dstBucket, but no subBucket value",
+			"srcBucket",
+			"subBucket",
+			"this is subBucket key",
+			"this is subBucket value",
+			"dstBucket",
+			true,
+			false,
+			false,
+			true,
+			berrors.ErrIncompatibleValue,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			db := btesting.MustCreateDB(t)
+
+			// arrange
+			if err := db.Update(func(tx *bolt.Tx) error {
+				// create src bucket
+				srcBucket, bErr := tx.CreateBucketIfNotExists([]byte(tc.srcBucketName))
+				if bErr != nil {
+					t.Fatalf("error creating src bucket %s: %v", tc.srcBucketName, bErr)
+				}
+				// insert K/V pair into src bucket
+				if pErr := srcBucket.Put([]byte("foo"), []byte("0000")); pErr != nil {
+					t.Fatal(pErr)
+				}
+
+				if tc.subBucketExistSrcBucket {
+					// create subBucket within srcBucket
+					subBucket, sErr := srcBucket.CreateBucket([]byte(tc.subBucketName))
+					if sErr != nil {
+						t.Fatal(sErr)
+					}
+					// insert K/V pair into  subBucket
+					if pErr := subBucket.Put([]byte(tc.subBucketKey), []byte(tc.subBucketValue)); pErr != nil {
+						t.Fatal(pErr)
+					}
+				}
+
+				if tc.keyNoSubBucketSrcBucket {
+					if pErr := srcBucket.Put([]byte(tc.subBucketName), []byte(tc.subBucketValue)); pErr != nil {
+						t.Fatal(pErr)
+					}
+				}
+
+				// create dst bucket
+				dstBucket, bErr := tx.CreateBucketIfNotExists([]byte(tc.dstBucketName))
+				if bErr != nil {
+					t.Fatalf("error creating dst bucket %s: %v", tc.dstBucketName, bErr)
+				}
+				// insert K/V pair into dst bucket
+				if pErr := dstBucket.Put([]byte("bar"), []byte("0000")); pErr != nil {
+					t.Fatal(pErr)
+				}
+
+				if tc.subBucketExistDstBucket {
+					// create subBucket within dstBucket
+					_, sErr := dstBucket.CreateBucket([]byte(tc.subBucketName))
+					if sErr != nil {
+						t.Fatal(sErr)
+					}
+				}
+
+				if tc.keyNoSubBucketDstBucket {
+					if pErr := dstBucket.Put([]byte(tc.subBucketName), []byte(tc.subBucketValue)); pErr != nil {
+						t.Fatal(pErr)
+					}
+				}
+
+				return nil
+			}); err != nil {
+				t.Fatal(err)
+			}
+			db.MustCheck()
+
+			// act: move the subBucket from srcBucket to dstBucket
+			if err := db.Update(func(tx *bolt.Tx) error {
+				srcBucket := tx.Bucket([]byte(tc.srcBucketName))
+				if srcBucket == nil {
+					t.Fatalf("src bucket %s does not exist: %v", tc.srcBucketName, berrors.ErrBucketNotFound)
+				}
+
+				dstBucket := tx.Bucket([]byte(tc.dstBucketName))
+				if dstBucket == nil {
+					t.Fatalf("dst bucket %s does not exist: %v", tc.dstBucketName, berrors.ErrBucketNotFound)
+				}
+
+				mvErr := tx.MoveSubBucket([]byte(tc.subBucketName), srcBucket, dstBucket)
```

Comment:
@ahrtr @fuweid  i have used it here 

I think it is better to keep this method, so we can create `move-bucket` cmd later on, wdyt ?